artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/broker/AMQPLargeMessage.java

```diff
@@ -126,6 +127,20 @@ public void closeLargeMessage() throws Exception {
       parsingData = null;
    }
 
+   public synchronized void releaseEncodedBuffer() {
+      if (temporaryBuffer != null && temporaryBuffer.release()) {
+         temporaryBuffer = null;
+      }
+   }
+
+   public synchronized ByteBuf getSavedEncodeBuffer() {
+      if (temporaryBuffer == null) {
+         temporaryBuffer = PooledByteBufAllocator.DEFAULT.buffer(getEstimateSavedEncode());
+         saveEncoding(temporaryBuffer);
+      }
+      return temporaryBuffer.retain();
```

Comment:
retain is a costly operation, is it necessary? Or we just document that the saved encode buffer cannot be conserved but just immediately used?
I would document it anyway to save a user to forget to release it and cause memory leaks

Reply:
Retain is just a ref counter increment.

Reply:
I know but looking at the naming used on Netty I think is a good idea to communicate when something is to be relased after usage and when not, just to be less error prone...

Reply:
I renamed the method. This is not much my style of coding. I have seen other code you've written and you would create a new method.  but I will submit and created the new method now.

pushed it ! :)